diff --git a/dist/src/ember.js b/dist/src/ember.js
index 2fd36fa936e5c7ac499b5ce2d9d2e0af74666292..f7acebd6612d21ab28fb90358c3177c5482b182b 100644
--- a/dist/src/ember.js
+++ b/dist/src/ember.js
@@ -41,12 +41,12 @@ export function ember() {
                 if (!config.optimizeDeps.esbuildOptions) {
                     config.optimizeDeps.esbuildOptions = {};
                 }
-                if (config.optimizeDeps.esbuildOptions.plugins) {
-                    config.optimizeDeps.esbuildOptions.plugins.push(esBuildResolver());
-                }
-                else {
-                    config.optimizeDeps.esbuildOptions.plugins = [esBuildResolver()];
-                }
+                // if (config.optimizeDeps.esbuildOptions.plugins) {
+                //     config.optimizeDeps.esbuildOptions.plugins.push(esBuildResolver());
+                // }
+                // else {
+                //     config.optimizeDeps.esbuildOptions.plugins = [esBuildResolver()];
+                // }
                 if (!config.build) {
                     config.build = {};
                 }
@@ -72,8 +72,8 @@ export function ember() {
                 // vite will try to transpile away typescript in .ts files using
                 // esbuild. But if we have any typescript, we expect it to get handled
                 // by babel, because we don't want esbuild's decorator implementation.
-                if (config.esbuild == null) {
-                    config.esbuild = false;
+                if (config.oxc == null) {
+                    config.oxc = false;
                 }
                 minification(config, env.mode);
             },
