diff --git a/pkg/standalone.js b/pkg/standalone.js
index 7be08fc..35777bf 100644
--- a/pkg/standalone.js
+++ b/pkg/standalone.js
@@ -1,7 +1,8 @@
-import init from "./standalone/content_tag.js";
+import { initSync } from "./standalone/content_tag.js";
+import module from "./standalone/content_tag_bg.wasm";
 import { Preprocessor as WasmPreprocessor } from "./standalone/content_tag.js";
 
-await init();
+initSync(module);
 
 const defaultOptions = {
   inline_source_map: false,
diff --git a/pkg/standalone/content_tag.js b/pkg/standalone/content_tag.js
index aaefe00..bb20026 100644
--- a/pkg/standalone/content_tag.js
+++ b/pkg/standalone/content_tag.js
@@ -20,6 +20,7 @@ function takeObject(idx) {
     return ret;
 }
 
+const { TextDecoder, TextEncoder } = require('util');
 const cachedTextDecoder = (typeof TextDecoder !== 'undefined' ? new TextDecoder('utf-8', { ignoreBOM: true, fatal: true }) : { decode: () => { throw Error('TextDecoder not available') } } );
 
 if (typeof TextDecoder !== 'undefined') { cachedTextDecoder.decode(); };
