diff --git a/node_modules/decorator-transforms/dist/index.js b/node_modules/decorator-transforms/dist/index.js
index fce3aeb..c23d8e4 100644
--- a/node_modules/decorator-transforms/dist/index.js
+++ b/node_modules/decorator-transforms/dist/index.js
@@ -4,10 +4,13 @@ import {
 import "./chunk-CSAU5B4Q.js";
 
 // src/index.ts
-import { createRequire } from "module";
+
 import { ImportUtil } from "babel-import-util";
-var req = createRequire(import.meta.url);
-var { default: decoratorSyntax } = req("@babel/plugin-syntax-decorators");
+
+// https://github.com/ef4/decorator-transforms/pull/27
+import PluginSyntaxDecorators from "@babel/plugin-syntax-decorators";
+const decoratorSyntax = PluginSyntaxDecorators.default || PluginSyntaxDecorators;
+
 function makeVisitor(babel) {
   const t = babel.types;
   return {
