diff --git a/dist/index.js b/dist/index.js
index a28eb37cf3d1d52ce6042881556b44fe67e0f355..3938062df30cebb58be220bc3dab03c61f1835ad 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -4,10 +4,13 @@ import {
 import "./chunk-CSAU5B4Q.js";
 
 // src/index.ts
-import { createRequire } from "module";
+
 import { ImportUtil } from "babel-import-util";
-var req = createRequire(import.meta.url);
-var { default: decoratorSyntax } = req("@babel/plugin-syntax-decorators");
+
+// https://github.com/ef4/decorator-transforms/pull/27
+import PluginSyntaxDecorators from "@babel/plugin-syntax-decorators";
+const decoratorSyntax = PluginSyntaxDecorators.default || PluginSyntaxDecorators;
+
 function makeVisitor(babel) {
   const t = babel.types;
   return {
