diff --git a/node_modules/ember-cli/lib/models/project.js b/node_modules/ember-cli/lib/models/project.js
index 7b3c7c9..e0c5fb9 100644
--- a/node_modules/ember-cli/lib/models/project.js
+++ b/node_modules/ember-cli/lib/models/project.js
@@ -716,6 +716,11 @@ class Project {
     logger.info('getProjectRoot %s -> %s', process.cwd(), directory);
     return directory;
   }
+
+  // Allows us to use recent ember-cli with ember-source 3.28.12
+  bowerDependencies(){
+    return [];
+  }
 }
 
 class NotFoundError extends Error {
