diff --git a/dist/ember.debug.js b/dist/ember.debug.js
index d205f434b0dd2bbb124ddcc62d6e2284b3910f32..e2044b16e7df230ff31a9df1703ffb22c9dad3d0 100644
--- a/dist/ember.debug.js
+++ b/dist/ember.debug.js
@@ -9298,7 +9298,7 @@ var define, require;
           // Ideally, we'd use MutableArray.detect but for unknown reasons this causes
           // the node tests to fail strangely.
           function isMutableArray(obj) {
-            return obj != null && typeof obj.replace === 'function';
+            return obj != null && typeof obj.replace === 'function' && !Array.isArray(obj);
           }
           function replace(array, start, deleteCount, items = EMPTY_ARRAY$3) {
             if (isMutableArray(array)) {
diff --git a/dist/ember.prod.js b/dist/ember.prod.js
index 93026b57ac2527ab05a483c168211318b039a7c9..76a03c798f5226a7fd3de69c114381b559bdae0a 100644
--- a/dist/ember.prod.js
+++ b/dist/ember.prod.js
@@ -8042,7 +8042,7 @@ var define, require;
           // Ideally, we'd use MutableArray.detect but for unknown reasons this causes
           // the node tests to fail strangely.
           function isMutableArray(obj) {
-            return obj != null && typeof obj.replace === 'function';
+            return obj != null && typeof obj.replace === 'function' && !Array.isArray(obj);
           }
           function replace(array, start, deleteCount, items = EMPTY_ARRAY$3) {
             if (isMutableArray(array)) {
diff --git a/dist/packages/shared-chunks/array-BGH9y76r.js b/dist/packages/shared-chunks/array-BGH9y76r.js
index 47fa056e935d4ddeae71a9a2d9ad801c4df21195..a8140f87cd68ab678a69a32921bcf70e93839b3f 100644
--- a/dist/packages/shared-chunks/array-BGH9y76r.js
+++ b/dist/packages/shared-chunks/array-BGH9y76r.js
@@ -67,7 +67,7 @@ const EMPTY_ARRAY = Object.freeze([]);
 // Ideally, we'd use MutableArray.detect but for unknown reasons this causes
 // the node tests to fail strangely.
 function isMutableArray(obj) {
-  return obj != null && typeof obj.replace === 'function';
+  return obj != null && typeof obj.replace === 'function' && !Array.isArray(obj);
 }
 function replace(array, start, deleteCount, items = EMPTY_ARRAY) {
   if (isMutableArray(array)) {
