diff --git a/dist/packages/@ember/-internals/deprecations/index.js b/dist/packages/@ember/-internals/deprecations/index.js
index 894cd5c5e28db30f8798ee828b8364c94d488964..bffb6b1c0d22d3611f8e85414c0f34f0e416e4f4 100644
--- a/dist/packages/@ember/-internals/deprecations/index.js
+++ b/dist/packages/@ember/-internals/deprecations/index.js
@@ -123,7 +123,7 @@ function deprecateUntil(message, deprecation) {
   if (deprecation.isRemoved) {
     throw new Error(`The API deprecated by ${options.id} was removed in ember-source ${options.until}. The message was: ${message}. Please see ${options.url} for more details.`);
   }
-  (isDevelopingApp() && !(deprecation.test) && deprecate(message, deprecation.test, options));
+  (true && !(deprecation.test) && deprecate(message, deprecation.test, options));
 }
 const {
   EXTEND_PROTOTYPES
diff --git a/dist/packages/@ember/debug/lib/deprecate.js b/dist/packages/@ember/debug/lib/deprecate.js
index 2a40f7cca607ab80beb267d79243fc30f16c92a1..9c15e205f3fb1661b8f1ae397c2338c42a3cc601 100644
--- a/dist/packages/@ember/debug/lib/deprecate.js
+++ b/dist/packages/@ember/debug/lib/deprecate.js
@@ -50,7 +50,7 @@ let missingOptionsDeprecation;
 let missingOptionsIdDeprecation;
 let missingOptionDeprecation = () => '';
 let deprecate = () => {};
-if (isDevelopingApp()) {
+if (true) {
   registerHandler = function registerHandler(handler) {
     registerHandler$1('deprecate', handler);
   };
diff --git a/dist/packages/@ember/debug/lib/handlers.js b/dist/packages/@ember/debug/lib/handlers.js
index cd68afe479229ed77b4e4c93759aa90913cdc49c..6e7d71bbeb28f18259fa8afb48ef78279f25c531 100644
--- a/dist/packages/@ember/debug/lib/handlers.js
+++ b/dist/packages/@ember/debug/lib/handlers.js
@@ -3,7 +3,7 @@ import { isDevelopingApp } from '@embroider/macros';
 let HANDLERS = {};
 let registerHandler = function registerHandler(_type, _callback) {};
 let invoke = () => {};
-if (isDevelopingApp()) {
+if (true) {
   registerHandler = function registerHandler(type, callback) {
     let nextHandler = HANDLERS[type] || (() => {});
     HANDLERS[type] = (message, options) => {
diff --git a/dist/packages/shared-chunks/array-BGH9y76r.js b/dist/packages/shared-chunks/array-BGH9y76r.js
index 47fa056e935d4ddeae71a9a2d9ad801c4df21195..a8140f87cd68ab678a69a32921bcf70e93839b3f 100644
--- a/dist/packages/shared-chunks/array-BGH9y76r.js
+++ b/dist/packages/shared-chunks/array-BGH9y76r.js
@@ -67,7 +67,7 @@ const EMPTY_ARRAY = Object.freeze([]);
 // Ideally, we'd use MutableArray.detect but for unknown reasons this causes
 // the node tests to fail strangely.
 function isMutableArray(obj) {
-  return obj != null && typeof obj.replace === 'function';
+  return obj != null && typeof obj.replace === 'function' && !Array.isArray(obj);
 }
 function replace(array, start, deleteCount, items = EMPTY_ARRAY) {
   if (isMutableArray(array)) {
