diff --git a/script/bulk_import/vbulletin.rb b/script/bulk_import/vbulletin.rb
index 9b3d7315e8..3ec3414ecc 100644
--- a/script/bulk_import/vbulletin.rb
+++ b/script/bulk_import/vbulletin.rb
@@ -55,30 +55,31 @@ class BulkImport::VBulletin < BulkImport::Base
     SiteSetting.clean_up_uploads = false
     SiteSetting.clean_orphan_uploads_grace_period_hours = 43200
 
-    import_groups
-    import_users
-    import_group_users
+    #import_groups
+    #import_users
+    #import_group_users
 
-    import_user_emails
-    import_user_stats
+    #import_user_emails
+    #import_user_stats
 
-    import_user_passwords
-    import_user_salts
-    import_user_profiles
+    #import_user_passwords
+    #import_user_salts
+    #import_user_profiles
 
-    import_categories
-    import_topics
-    import_posts
+    #import_categories
+    #import_topics
+    #import_posts
 
-    import_likes
+    #import_likes
 
-    import_private_topics
-    import_topic_allowed_users
-    import_private_posts
+    #import_private_topics
+    #import_topic_allowed_users
+    #import_private_posts
 
-    create_permalink_file
-    import_attachments
-    import_avatars
+    create_permalinks
+    #create_permalink_file
+    #import_attachments
+    #import_avatars
     #import_signatures
   end
 
@@ -106,7 +107,7 @@ class BulkImport::VBulletin < BulkImport::Base
     puts "Importing users..."
 
     users = mysql_stream <<-SQL
-        SELECT u.userid, username, email, joindate, birthday, ipaddress, u.usergroupid, bandate, liftdate, usertitle
+        SELECT u.userid, username, email, joindate, birthday, ipaddress, u.usergroupid, bandate, liftdate, u.usertitle
           FROM #{TABLE_PREFIX}user u
      LEFT JOIN #{TABLE_PREFIX}userban ub ON ub.userid = u.userid
          WHERE u.userid > #{@last_imported_user_id}
@@ -488,6 +489,73 @@ class BulkImport::VBulletin < BulkImport::Base
     end
   end
 
+  def create_permalinks
+    puts '', 'Creating permalinks...', ''
+
+
+    puts '    User pages...'
+
+
+    start = Time.now
+    count = 0
+    now = Time.zone.now
+
+
+    sql = "COPY permalinks (url, created_at, updated_at, external_url) FROM STDIN"
+
+
+    @raw_connection.copy_data(sql, @encoder) do
+      User.includes(:_custom_fields).find_each do |u|
+        count += 1
+        ucf = u.custom_fields
+        if ucf && ucf["import_id"]
+          @raw_connection.put_copy_data(
+            ["sdmb/member.php?u=#{ucf["import_id"]}", now, now, "/users/#{u.username}"]
+          )
+        end
+
+
+        print "\r%7d - %6d/sec" % [count, count.to_f / (Time.now - start)] if count % 5000 == 0
+      end
+    end
+
+
+    puts '', '', '    Topics and posts...'
+
+
+    start = Time.now
+    count = 0
+
+
+    sql = "COPY permalinks (url, topic_id, post_id, created_at, updated_at) FROM STDIN"
+
+
+    @raw_connection.copy_data(sql, @encoder) do
+      Post.includes(:_custom_fields).find_each do |post|
+        count += 1
+        pcf = post.custom_fields
+        if pcf && pcf["import_id"]
+          topic = post.topic
+          if topic.present?
+            tcf = topic.custom_fields
+            if post.post_number == 1
+              @raw_connection.put_copy_data(
+                ["sdmb/showthread.php?t=#{tcf["import_id"]}", topic.id, nil, now, now]
+              )
+            else
+              @raw_connection.put_copy_data(
+                ["sdmb/showpost.php?p=#{pcf["import_id"]}", nil, post.id, now, now]
+              )
+            end
+          end
+        end
+
+
+        print "\r%7d - %6d/sec" % [count, count.to_f / (Time.now - start)] if count % 5000 == 0
+      end
+    end
+  end
+
   def create_permalink_file
     puts '', 'Creating Permalink File...', ''
 
