diff --git a/node_modules/@ember/jquery/vendor/jquery/component.dollar.js b/node_modules/@ember/jquery/vendor/jquery/component.dollar.js
index 8c0cfd5..4232e56 100644
--- a/node_modules/@ember/jquery/vendor/jquery/component.dollar.js
+++ b/node_modules/@ember/jquery/vendor/jquery/component.dollar.js
@@ -1,5 +1,5 @@
+import Ember from 'ember';
 import { assert, deprecate } from '@ember/debug';
-import EmberObject from '@ember/object';
 import Component from '@ember/component';
 
 (function() {
@@ -8,7 +8,7 @@ import Component from '@ember/component';
    * base class to be reopened without triggering the
    * ember.component.reopen deprecation in Ember itself.
    */
-  EmberObject.reopen.call(Component, {
+  Ember.Object.reopen.call(Component, {
     $(sel) {
       assert(
         "You cannot access this.$() on a component with `tagName: ''` specified.",
@@ -28,6 +28,8 @@ import Component from '@ember/component';
       );
 
       if (this.element) {
+        // same technique the shim uses
+        let jQuery = self['jQuery'];
         return sel ? jQuery(sel, this.element) : jQuery(this.element);
       }
     }
