diff --git a/node_modules/content-tag/content_tag.js b/node_modules/content-tag/content_tag.js
index e9be50d..9e20da2 100644
--- a/node_modules/content-tag/content_tag.js
+++ b/node_modules/content-tag/content_tag.js
@@ -448,11 +448,17 @@ module.exports.__wbindgen_memory = function() {
     return addHeapObject(ret);
 };
 
-const path = require('path').join(__dirname, 'content_tag_bg.wasm');
-const bytes = require('fs').readFileSync(path);
-
-const wasmModule = new WebAssembly.Module(bytes);
-const wasmInstance = new WebAssembly.Instance(wasmModule, imports);
-wasm = wasmInstance.exports;
-module.exports.__wasm = wasm;
+// Check for nodejs environment
+if (process.version) {
+  const path = require('path').join(__dirname, 'content_tag_bg.wasm');
+  const bytes = require('fs').readFileSync(path);
+
+  const wasmModule = new WebAssembly.Module(bytes);
+  const wasmInstance = new WebAssembly.Instance(wasmModule, imports);
+  wasm = wasmInstance.exports;
+} else {
+  const load = require('./content_tag_bg.wasm').default;
+  wasm = load(imports);
+  module.exports.__wasm = wasm;
+}
 
