diff --git a/node_modules/content-tag/pkg/standalone.js b/node_modules/content-tag/pkg/standalone.js
index 0df00d7..c19b607 100644
--- a/node_modules/content-tag/pkg/standalone.js
+++ b/node_modules/content-tag/pkg/standalone.js
@@ -1,5 +1,6 @@
-import init from "./standalone/content_tag.js";
+import { initSync } from "./standalone/content_tag.js";
+import module from "./standalone/content_tag_bg.wasm";
 
-await init();
+initSync(module);
 
 export { Preprocessor } from "./standalone/content_tag.js";
diff --git a/node_modules/content-tag/pkg/standalone/content_tag.js b/node_modules/content-tag/pkg/standalone/content_tag.js
index 70545cf..e6eb36b 100644
--- a/node_modules/content-tag/pkg/standalone/content_tag.js
+++ b/node_modules/content-tag/pkg/standalone/content_tag.js
@@ -20,6 +20,7 @@ function takeObject(idx) {
     return ret;
 }
 
+const { TextDecoder, TextEncoder } = require('util');
 const cachedTextDecoder = (typeof TextDecoder !== 'undefined' ? new TextDecoder('utf-8', { ignoreBOM: true, fatal: true }) : { decode: () => { throw Error('TextDecoder not available') } } );
 
 if (typeof TextDecoder !== 'undefined') { cachedTextDecoder.decode(); };
