diff --git a/node_modules/ember-template-imports/lib/index.js b/node_modules/ember-template-imports/lib/index.js
index 42dd417..2d10552 100644
--- a/node_modules/ember-template-imports/lib/index.js
+++ b/node_modules/ember-template-imports/lib/index.js
@@ -7,3 +7,7 @@ Object.defineProperty(exports, "DEFAULT_PARSE_TEMPLATES_OPTIONS", { enumerable:
 var util_1 = require("./util");
 Object.defineProperty(exports, "isStrictMode", { enumerable: true, get: function () { return util_1.isStrictMode; } });
 Object.defineProperty(exports, "isSupportedScriptFileExtension", { enumerable: true, get: function () { return util_1.isSupportedScriptFileExtension; } });
+var babel_plugin_1 = require("../src/babel-plugin");
+Object.defineProperty(exports, "babelPlugin", { enumerable: true, get: function () { return babel_plugin_1; } });
+var preprocess_embedded_templates1 = require("./preprocess-embedded-templates");
+Object.defineProperty(exports, "preprocessEmbeddedTemplates", { enumerable: true, get: function () { return preprocess_embedded_templates1.preprocessEmbeddedTemplates; } });
diff --git a/node_modules/ember-template-imports/lib/preprocess-embedded-templates.js b/node_modules/ember-template-imports/lib/preprocess-embedded-templates.js
index f50518b..9c49386 100644
--- a/node_modules/ember-template-imports/lib/preprocess-embedded-templates.js
+++ b/node_modules/ember-template-imports/lib/preprocess-embedded-templates.js
@@ -27,8 +26,12 @@ function replacementFrom(template, index, oldLength, newLength, type) {
     };
 }
 function loadGetTemplateLocals(path, exportPath) {
-    // eslint-disable-next-line @typescript-eslint/no-var-requires
-    const templateLocals = require(path);
+    let templateLocals = path;
+
+    if (typeof path === "string") {
+      // eslint-disable-next-line @typescript-eslint/no-var-requires
+      templateLocals = require(path);
+    }
     let getTemplateLocals = templateLocals;
     for (const segment of exportPath.split('.')) {
         getTemplateLocals = getTemplateLocals[segment];
