diff --git a/node_modules/ember-this-fallback/lib/helpers/string.js b/node_modules/ember-this-fallback/lib/helpers/string.js
index c6f4e65..91ed2af 100644
--- a/node_modules/ember-this-fallback/lib/helpers/string.js
+++ b/node_modules/ember-this-fallback/lib/helpers/string.js
@@ -21,6 +21,8 @@ function squish(str) {
 }
 exports.squish = squish;
 function classify(str) {
-    return (0, lodash_1.upperFirst)((0, lodash_1.camelCase)(str));
+    const parts = str.split('/');
+    const classifiedParts = parts.map((p) => (0, lodash_1.upperFirst)((0, lodash_1.camelCase)(p)));
+    return classifiedParts.join('::');
 }
 exports.classify = classify;
