diff --git a/node_modules/ember-cli/lib/models/project.js b/node_modules/ember-cli/lib/models/project.js
index b21a9bcd4..6537fcc2e 100644
--- a/node_modules/ember-cli/lib/models/project.js
+++ b/node_modules/ember-cli/lib/models/project.js
@@ -409,6 +409,7 @@ class Project {
     @return {Object}       Bower dependencies
    */
   bowerDependencies(bower) {
+    return {}; // Silence deprecation until Discourse is on ember-source 4.x
     deprecate(
       [
         '`bowerDependencies` has been deprecated.',
